build/lib/src/asset/reader.dart

```diff
@@ -8,16 +8,27 @@ import 'package:glob/glob.dart';
 
 import 'id.dart';
 
-/// Abstract interface for reading assets.
+/// Standard interface for reading an asset within in a package.
+///
+/// An [AssetReader] is required when implementing the `runBuilder` command.
 abstract class AssetReader {
+  /// Returns a [Future] that completes with the bytes of a binary asset.
+  ///
+  /// * Throws a [PackageNotFoundException] if [id.package] is not found.
```

Comment:
Acknowledged. I'm going to try and add an e2e test in a future PR and I'll try it out.